packages/twenty-server/src/workspace/workspace-query-runner/factories/arg-setters.factory.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable } from '@nestjs/common';
+
+import { FieldMetadataInterface } from 'src/metadata/field-metadata/interfaces/field-metadata.interface';
+import { WorkspaceQueryRunnerOptions } from 'src/workspace/workspace-query-runner/interfaces/query-runner-option.interface';
+import { ObjectMetadataInterface } from 'src/metadata/field-metadata/interfaces/object-metadata.interface';
+
+import { WorkspaceDataSourceService } from 'src/workspace/workspace-datasource/workspace-datasource.service';
+import { FieldMetadataType } from 'src/metadata/field-metadata/field-metadata.entity';
+
+@Injectable()
+export class ArgsSettersFactory {
+  constructor(
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+  ) {}
+
+  async create(
+    args: Record<string, any>,
+    options: WorkspaceQueryRunnerOptions,
+  ) {
+    const fieldMetadataCollection = options.fieldMetadataCollection;
+
+    const fieldMetadataMap = new Map(
+      fieldMetadataCollection.map((fieldMetadata) => [
+        fieldMetadata.name,
+        fieldMetadata,
+      ]),
+    );
+
+    return this.applySettersToArgsRecursive(args, options, fieldMetadataMap);
+  }
+
+  private async applySettersToArgsRecursive(
+    args: Record<string, any>,
+    options: WorkspaceQueryRunnerOptions,
+    fieldMetadataMap: Map<string, FieldMetadataInterface>,
+  ) {
+    // If it's not an object, we don't need to do anything
+    if (typeof args !== 'object' || args === null) {
+      return args;
+    }
+
+    // If it's an array, we need to map all items
+    if (Array.isArray(args)) {
+      return Promise.all(
+        args.map((arg) =>
+          this.applySettersToArgsRecursive(arg, options, fieldMetadataMap),
+        ),
+      );
+    }
+
+    const newArgs = {};
+
+    for (const [key, value] of Object.entries(args)) {
+      const fieldMetadata = fieldMetadataMap.get(key);
+
+      if (!fieldMetadata) {
+        newArgs[key] = await this.applySettersToArgsRecursive(
+          value,
+          options,
+          fieldMetadataMap,
+        );
+        continue;
+      }
+
+      switch (fieldMetadata.type) {
+        case FieldMetadataType.POSITION:
+          newArgs[key] = await this.buildPositionValue(
+            value,
+            options.objectMetadataItem,
+            options.workspaceId,
+          );
+          break;
+        default:
+          newArgs[key] = await this.applySettersToArgsRecursive(
+            value,
+            options,
+            fieldMetadataMap,
+          );
+      }
+    }
+
+    return newArgs;
```

Comment:
We can also run this in parallel, like you did for an array.
```suggestion
    const tasks = Object.entries(args).map(async ([key, value]) => {
      const fieldMetadata = fieldMetadataMap.get(key);
      if (!fieldMetadata) {
        return [key, await this.applySettersToArgsRecursive(value, options, fieldMetadataMap, dataSourceService)];
      }

      switch (fieldMetadata.type) {
        case FieldMetadataType.POSITION:
          return [key, await this.buildPositionValue(value, options.objectMetadataItem, options.workspaceId, dataSourceService)];
        default:
          return [key, await this.applySettersToArgsRecursive(value, options, fieldMetadataMap, dataSourceService)];
      }
    });

    const newArgsEntries = await Promise.all(tasks);

    return Object.fromEntries(newArgsEntries);
```